mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp

```diff
@@ -186,11 +186,35 @@ static void checkSafeToTileToForall(TilingInterface op,
   }
 }
 
+/// Returns true if `size` is dynamic multiplication of `stride`.
```

Comment:
Not sure I follow. The `tileDividesIterationDomain` is checking for (ub - lb) % tilesize = 0. Why doesnt that catch the case you are trying to cover here?

Reply:
@MaheshRavishankar 
you are right, my change adds dynamic support for `(ub - lb) % tilesize` where `lb` is zero, and ub is dynamicly defined by `possibly_dynamic_ub = possibly_dynamic_tilesize * possibly_dynamic_stride`.
Some background for this change - I am using split on dynamic operations (i.e. `linalg::generic`) where the I use `arith.divui` to calculate a tile which fits my cache and another part of `arith.remui` if the dynamic size won't be perfectly divided.
After the split, the first part size is defined by `arith.muli` where one of its operands is the tile size I am using to tile the operation - but the the `ub` is now considered dynamic although we can know the tile size divides it.

Reply:
Ok, I see what you are trying. I am not a fan of this kind of piecemeal approaches. I tried something. I changed your input to use affine maps. Still didnt work. I get something like this
```
%1 = affine.apply affine_map<()[s0] -> ((s0 floordiv 80) * 80)>()[%dim]
%3 = scf.for %arg2 = %c0 to %1 step %c80 iter_args(%arg3 = %extracted_slice_1) -> (tensor<?xi16>) {
      %5 = affine.min affine_map<(d0)[s0] -> (-d0 + (s0 floordiv 80) * 80, 80)>(%arg2)[%dim]
```

IIRC adding `--scf-for-loop-canonicalization` should have fixed the problem, but it didnt

cc @matthias-springer  I thought this was fixed at some time.

Anyway, I am OK with this patch overall given where things are, but this needs to account for the lower bound of the loop as well.

Reply:
I haven't looked into this pass for a long time. What I vaguely remember is that `-scf-for-loop-canonicalization` has always been a bit fragile due to limitations around semi-affine expressions in `FlatAffineValueConstraints`.

Reply:
@MaheshRavishankar thanks for the replay!
I tried to take a look at `scf-for-loop-canonicalization` which I was unfamiliar with, as for now I didn't manage to find the issue why it does not manage to remove the `affine.min` in this case. I updated the MR to also handle none `0` offset.

Reply:
@MaheshRavishankar can you please take a look on the latest patch?